velox/functions/prestosql/Comparisons.cpp

```diff
@@ -97,9 +97,10 @@ struct SimdComparator {
   template <
       TypeKind kind,
       typename std::enable_if_t<
-          xsimd::has_simd_register<
-              typename TypeTraits<kind>::NativeType>::value &&
-              kind != TypeKind::BOOLEAN,
+          (xsimd::has_simd_register<
```

Comment:
Extract this condition out so we can reuse it in different specialization of the template

Reply:
I will be reverting this change in #4434 as well. Do you still want me to extract this condition?

Reply:
That's good then